internal/store/common.go

```diff
@@ -0,0 +1,37 @@
+package store
+
+import (
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+)
+
+func BuildBaseListQuery(db *gorm.DB, orgId uuid.UUID, labels map[string]string) *gorm.DB {
+	query := db.Where("org_id = ?", orgId).Order("name")
+	query = LabelSelectionQuery(query, labels)
+	return query
+}
+
+func AddPaginationToQuery(query *gorm.DB, limit int, cont string) *gorm.DB {
+	if limit > 0 {
+		query = query.Limit(limit)
+	}
+	if cont != "" {
+		query = query.Where("name > ?", cont)
+	}
+
+	return query
+}
+
+func FormatResourcesAfterPagination(query *gorm.DB, limit int, lastItemName string) (*string, *int64) {
+	if limit == 0 {
+		return nil, nil
+	}
+
+	var count int64
+	query.Where("name > ?", lastItemName).Count(&count)
```

Comment:
This is expensive.

https://dba.stackexchange.com/questions/317626/optimizing-select-count-result-in-postgresql
https://wiki.postgresql.org/wiki/Slow_Counting

Apparently, the DB needs to walk all the rows for every query you make; if you keep iterating as you query this is an O(n!) complexity 

I propose we don't return a count, and we let iteratively discover. Alternatively, I think we may need to cache the count and provide an approximate count to avoid the O(n!) trap.

Reply:
We will probably be out of memory before we return millions of results. 
We can maybe try to enforce pagination? AWS is doing that, they limit the result to 1K objects.

Reply:
I agree that we should limit to 1K objects.  Enforcing that later would be an API breakage.  @mangelajo WDYT?

Reply:
Yeah, it should be limited. My worry is the count result, we can either cache it for subsequent pagination requests, or as discussed in slack, accept the previous count as the follow up pagination requests.

Copying here your description of the idea which I think is good: 

![image](https://github.com/flightctl/flightctl/assets/1176434/21c6ac5a-4247-4307-a4f8-301aa870c101)